WordPress/src/main/java/org/wordpress/android/ui/sitecreation/previews/SiteCreationPreviewFragment.kt

```diff
@@ -356,6 +413,91 @@ class SiteCreationPreviewFragment : SiteCreationBaseFormFragment(),
 
     private fun createFadeInAnimator(view: View) = ObjectAnimator.ofFloat(view, "alpha", 0f, 1f)
 
+    /**
+     * A simple helper to switch texts used to display progress of site being created.
+     * The helper is a runnable and one started (externally) will keep running until cancelled
+     * Each run changes the text to the next one in the sequence (stringIds) in a cyclic manner
+     *
+     * using a WeakReference for the text view will prevent future executions once the view is
+     * destroyed (or at least no longer referenced by fullscreenProgressLayout)
+     *
+     * @param textSwitcher - a week-ref to a text-switcher which displays the sequence
+     * @param stringIds - a list of string resource ids to sequence
+     * @param delay - how long to delay the next text-change (i.e. how long is each text displayed)
+     */
+    private inner class SiteCreationTextsProgressHelper(
+        val textSwitcher: WeakReference<TextSwitcher>,
+        val stringIds: List<Int>,
+        val delay: Long = SITE_CREATION_PREVIEW_TEXT_DURATION_MS
+    ) : Runnable {
+        val nText: Int = stringIds.size
+
+        // the count of how many times `nextText` was executed
+        var count = -1
+
+        var canceled = false
+
+        init {
+            textSwitcher.get()?.apply {
+                removeAllViews()
+                visibility = View.VISIBLE
+                setFactory {
+                    // inflate a text view that matches the design for this screen
+                    LayoutInflater.from(context)
+                            .inflate(R.layout.site_creation_progress_text, this, false)
+                }
+                // set fade in/out animations
+                inAnimation = AnimationUtils.loadAnimation(context, R.anim.fade_in)
```

Comment:
using existing fade-in/out animations (i.e. already defined in the project) @shaktimb - could you advise whether these animations are fine? they run for 600ms.